proto/src/main/java/com/google/gson/protobuf/ProtoTypeAdapter.java

```diff
@@ -190,6 +191,9 @@ public static Builder newBuilder() {
   private static final com.google.protobuf.Descriptors.FieldDescriptor.Type ENUM_TYPE =
       com.google.protobuf.Descriptors.FieldDescriptor.Type.ENUM;
 
+  private static final ConcurrentMap<String, Map<Class<?>, Method>> mapOfMapOfMethods =
+      new MapMaker().makeMap();
```

Comment:
Weak keys won't help generally, the Gson instance strongly references the classes its configured to encode and decode.